app/scripts/directives/overview/virtualMachineRow.js

```diff
@@ -87,15 +161,94 @@
       );
     };
     row.canStartOvm = function () {
-      return !isOvmRunning();
+      return !row.isOvmRunning();
     };
     row.canStopOvm = function () {
-      return isOvmRunning();
+      return row.isOvmRunning();
     };
     row.canRestartOvm = function () {
-      return isOvmRunning() &&
-             row.apiObject._vm &&
-             _.get(row.apiObject, '_pod.status.phase') === 'Running';
+      return row.isOvmInRunningPhase();
+    };
+    row.isWindowsVM = function () {
+      // https://github.com/kubevirt/kubevirt/blob/576d232e3c181134e69719cdb2d624ac52e7eecb/docs/devel/guest-os-info.md
+      var ovm = row.apiObject;
+      var vm = row.apiObject._vm;
+      var os = _.get(vm, 'metadata.labels["kubevirt.io/os"]') || _.get(ovm, 'metadata.labels["kubevirt.io/os"]');
+      return (os && _.startsWith(os, 'win'));
+    };
+    row.isRdpService = function () {
+      var ovm = row.apiObject;
+      return !_.isEmpty(ovm.services);
+    };
+
+    function rdpPortFinder(service) {
+      return _.find(_.get(service, 'spec.ports'), function (portObj) {
+        return portObj.targetPort === RDP_PORT;
+      });
+    }
+
+    function getAddressPort(service, ovm) {
+      var rdpPortObj = rdpPortFinder(service);
+      if (!rdpPortObj) {
+        return ;
+      }
+
+      var type = _.get(service, 'spec.type');
+      var port = _.get(rdpPortObj, "port");
+      var address;
+      switch (type) {
+        case 'LoadBalancer':
+          var externalIPs = _.get(service, 'spec.externalIPs');
+          if (_.isEmpty(externalIPs)) {
+            console.warn("externalIP is not defined for the LoadBalancer RDP Service: ", service, ovm);
+            return ;
+          }
+          address = externalIPs[0];
+          break;
+        case 'ClusterIP':
+          var clusterIP = _.get(service, 'spec.clusterIP');
```

Comment:
Suggest just assigning to `address` directly (here and below). Note that `var` is not block scope.

Reply:
done